Code/Framework/AzFramework/Tests/Spawnable/SpawnableScriptMediatorTests.cpp

```diff
@@ -0,0 +1,302 @@
+        }
+
+        virtual void OnDespawn(AzFramework::EntitySpawnTicket spawnTicket) override
+        {
+            AZStd::erase_if(
+                m_spawnedEntities,
+                [&spawnTicket](const TicketToEntitiesPair& pair) -> bool
+                {
+                    return pair.first == spawnTicket;
+                });
+        }
+
+        void SetUp() override
+        {
+            AllocatorsFixture::SetUp();
+
+            m_application = new TestApplication();
+            AZ::ComponentApplication::Descriptor descriptor;
+            m_application->Start(descriptor);
+
+            // Without this, the user settings component would attempt to save on finalize/shutdown. Since the file is
+            // shared across the whole engine, if multiple tests are run in parallel, the saving could cause a crash
+            // in the unit tests.
+            AZ::UserSettingsComponentRequestBus::Broadcast(&AZ::UserSettingsComponentRequests::DisableSaveOnFinalize);
+
+            auto managerInterface = AzFramework::SpawnableEntitiesInterface::Get();
+            m_manager = azrtti_cast<AzFramework::SpawnableEntitiesManager*>(managerInterface);
+        }
+
+        void TearDown() override
+        {
+            AzFramework::Scripts::SpawnableScriptNotificationsBus::MultiHandler::BusDisconnect();
+            // One more tick on the spawnable entities manager in order to delete the ticket fully.
+            while (m_manager->ProcessQueue(
+                       AzFramework::SpawnableEntitiesManager::CommandQueuePriority::High |
+                       AzFramework::SpawnableEntitiesManager::CommandQueuePriority::Regular) !=
+                   AzFramework::SpawnableEntitiesManager::CommandQueueStatus::NoCommandsLeft)
+                ;
+            
+            m_spawnedEntities = {};
+
+            delete m_application;
+            m_application = nullptr;
+
+            AllocatorsFixture::TearDown();
+        }
+
+        void WaitForResponse(AzFramework::Scripts::SpawnableScriptMediator& mediator)
+        {
+            for (size_t i=0; i<1000; ++i) // Don't do this indefinitely to avoid deadlocking on a failing test.
+            {
+                if (m_manager->ProcessQueue(
+                        AzFramework::SpawnableEntitiesManager::CommandQueuePriority::High |
+                        AzFramework::SpawnableEntitiesManager::CommandQueuePriority::Regular) ==
+                    AzFramework::SpawnableEntitiesManager::CommandQueueStatus::NoCommandsLeft)
+                {
+                    break;
+                }
+            }
+            mediator.OnTick(0, AZ::ScriptTimePoint());
+        }
+        
+        AzFramework::Scripts::SpawnableScriptAssetRef CreateSpawnable(size_t numElements)
+        {
+            auto spawnable = aznew AzFramework::Spawnable(
+                AZ::Data::AssetId::CreateString("{EB2E8A2B-F253-4A90-BBF4-55F2EED786B8}:0"), AZ::Data::AssetData::AssetStatus::Ready);
```

Comment:
Do we want to use AZStd::uuid to generate a random GUID? Or is using the same guid intentional? I see some places using multiple SpawnableScriptAssetRefs